### PR TITLE
Issue 469: Add dellhw_exporter support

### LINKED ISSUE #469: Add support for dellhw_exporter
It would be nice if this module supported deploying https://github.com/galexrt/dellhw_exporter

### PR DESCRIPTION
**Pull request by kkunkel**
I shamelessly copied and adjusted the puppetdb_exporter pull request.

#### Pull Request (PR) description
This PR adds support for https://github.com/galexrt/dellhw_exporter

#### This Pull Request (PR) fixes the following issues
Fixes #469

### CODE REVIEW

File: manifests/dellhw_exporter.pp [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+# @param service_name
+#  Name of the dellhw exporter service (default 'dellhw_exporter')
+# @param user
+#  User which runs the service
+# @param version
+#  The binary release version
+# @param omreport_path
+#  The file path to the omReport executable (default "/opt/dell/srvadmin/bin/omreport")
+class prometheus::dellhw_exporter (
+  String $download_extension              = 'tar.gz',
```
Comment (bastelfreak) [inline manifests/dellhw_exporter.pp]: for all the string datatypes where an empty string isnt allowed, can you please update it to String[1]?

Comment (kkunkel) [reply manifests/dellhw_exporter.pp]: sure, should I update the other modules? The lines you're referring to are a copy of lines from https://github.com/voxpupuli/puppet-prometheus/pull/463/files

Comment (bastelfreak) [reply manifests/dellhw_exporter.pp]: would be awesome if you like to do that!, but pleaee provide that change in another PR

Comment (bastelfreak) [reply manifests/dellhw_exporter.pp]: oh I missed it in my own PR, ups :(

Comment (kkunkel) [reply manifests/dellhw_exporter.pp]: Does this look better now?

Comment (bastelfreak) [reply manifests/dellhw_exporter.pp]: yes, thanks!

**Pull request merged**